ksql-api/src/main/java/io/confluent/ksql/api/server/BlockingQueryPublisher.java

```diff
@@ -132,7 +128,16 @@ protected void maybeSend() {
 
   @Override
   protected void afterSubscribe() {
-    queryHandle.start();
+    // Run async as it can block
+    executeOnWorker(queryHandle::start);
```

Comment:
This change to call `queryHandle::start` in a worker thread is unrelated to the addition of pull query support, right? (Just an improvement from the PR this one builds on?)

Reply:
Yep